source/base/multithread_info.cc

```diff
@@ -103,14 +103,43 @@ void MultithreadInfo::set_thread_limit(const unsigned int max_threads)
   Assert(n_max_threads==numbers::invalid_unsigned_int,
          ExcMessage("Calling set_thread_limit() more than once is not supported!"));
 
-  unsigned int max_threads_env = numbers::invalid_unsigned_int;
-  char *penv;
-  penv = getenv ("DEAL_II_NUM_THREADS");
-
-  if (penv!=NULL)
-    max_threads_env = Utilities::string_to_int(std::string(penv));
-
-  n_max_threads = std::min(max_threads, max_threads_env);
+  // set the maximal number of threads to the given value if one was specified
+  if (max_threads != numbers::invalid_unsigned_int)
```

Comment:
But we Assert that the value is ```invalid_unsigned_int``` a few lines above. There is no way it could have been set to anything else.